### DISCUSSION TITLE
Execute trigger with javascript

### DISCUSSION THREAD

**Discussion by joaosilva1002** [LiveHelperChat/livehelperchat#1898] [General]
Hi!

I'm trying to run different triggers based on the true or false result, but it displays this error, do you know how I could fix it?

![image](https://user-images.githubusercontent.com/93102624/215659215-b52fd498-845e-44d6-b0a7-c3ab386f84e4.png)

![image](https://user-images.githubusercontent.com/93102624/215659239-483c7b86-8f87-427e-9396-44a7ad71871e.png)

When I put it to display only the console log it returns me the way I need it, but I can't display it in the chat for the user

![image](https://user-images.githubusercontent.com/93102624/215659286-bbdb3a30-60f7-4c0c-93cc-2a9a1cf9a412.png)

![image](https://user-images.githubusercontent.com/93102624/215659305-92d15e47-221f-4e31-92d8-f0916b2bfc44.png)

**Comment by remdex** [comment]
You should be using this instead of JS approach. I highly recommend using that approach not JS to execute triggers...
https://doc.livehelperchat.com/docs/bot/check-conditions

And to execute javascript it should be `window.parent`

**Comment by joaosilva1002** [comment]
Initially I tried to do it that way, but the values ​​that are returned by my API are not calculated.

The calculation is done only through JavaScript.

I tried to follow this tutorial to run a trigger through javascript but it didn't work, not even with window.parent

https://doc.livehelperchat.com/docs/javascript-arguments#how-to-execute-any-trigger-manually

![image](https://user-images.githubusercontent.com/93102624/215849312-f008ab47-3f27-42a7-acda-63034011a26c.png)


Is there any other way I can force the execution of a trigger by JavaScript?

**Reply by remdex** [reply]
Window.parent....

But it's a bad practise you can juset set chat variable and compare it afterwards. In any case direct comparison in check for conditions to proceed also has to work. If not it's not so hard to extend in any case...
